### ISSUE TITLE
JS incompatability - Unexpected parsing behavior with implicit objects and switch-cases 

### ISSUE THREAD

**Issue by esthedebeste** [DanielXMoore/Civet#593]
```ts
switch x {
  case 2: {
    console.log("x is 2")
  }
}
```
compiles to
```ts
switch (x) {
  case {
    2: {
      log: console.log("x is 2"),
    },
  }:
}
```

(possibly just disable instantiating implicit objects in `case` matchers overall, `{} !== {}` so this will never be programmer intent)

**Comment by edemaine** [comment]
I believe the primary issue here is that `case (x): y` fails to parse, i.e., currently nothing is allowed after a `:` in a `case` environment. I believe we should allow "single-line" expressions here (including braced blocks like this), with `:` acting like `then`.

**Comment by edemaine** [comment]
Should be fixed in 0.6.17! Let us know if you run into any other forms that don't work.
